src/preamble.js

```diff
@@ -377,67 +368,74 @@ if (typeof SharedArrayBuffer === 'undefined' || typeof Atomics === 'undefined')
 #include "runtime_sab_polyfill.js"
 
 #if USE_PTHREADS
-#if !WASM
-if (typeof SharedArrayBuffer !== 'undefined') {
-  if (!ENVIRONMENT_IS_PTHREAD) buffer = new SharedArrayBuffer(INITIAL_TOTAL_MEMORY);
-} else {
-  if (!ENVIRONMENT_IS_PTHREAD) buffer = new ArrayBuffer(INITIAL_TOTAL_MEMORY);
-}
-updateGlobalBufferViews();
-
-#else
-if (!ENVIRONMENT_IS_PTHREAD) {
-#if ALLOW_MEMORY_GROWTH
-  wasmMemory = new WebAssembly.Memory({ 'initial': INITIAL_TOTAL_MEMORY / WASM_PAGE_SIZE , 'maximum': {{{ WASM_MEM_MAX }}} / WASM_PAGE_SIZE, 'shared': true });
-#else
-  wasmMemory = new WebAssembly.Memory({ 'initial': INITIAL_TOTAL_MEMORY / WASM_PAGE_SIZE , 'maximum': INITIAL_TOTAL_MEMORY / WASM_PAGE_SIZE, 'shared': true });
-#endif
-  buffer = wasmMemory.buffer;
-  assert(buffer instanceof SharedArrayBuffer, 'requested a shared WebAssembly.Memory but the returned buffer is not a SharedArrayBuffer, indicating that while the browser has SharedArrayBuffer it does not have WebAssembly threads support - you may need to set a flag');
-}
-
-updateGlobalBufferViews();
-#endif // !WASM
-#else // USE_PTHREADS
-
-// Use a provided buffer, if there is one, or else allocate a new one
-if (Module['buffer']) {
-  buffer = Module['buffer'];
-#if ASSERTIONS
-  assert(buffer.byteLength === INITIAL_TOTAL_MEMORY, 'provided buffer should be ' + INITIAL_TOTAL_MEMORY + ' bytes, but it is ' + buffer.byteLength);
```

Comment:
this assert seems to have been removed, was that intentional? or is it checked some other way?

Reply:
See line 429. In certain cases like growable memory the buffer size changes. Yes you can provide your own size with `Module['TOTAL_MEMORY']` but I just kind of make the assumption that if someone is providing their own buffer they know what they're doing without requiring them to make an extra step.